Sources/TSCUtility/InterruptHandler.swift

```diff
@@ -11,19 +11,19 @@
 import TSCLibc
 import TSCBasic
 
-/// Interrupt signal handling global variables
-private var wasInterrupted = false
-private var wasInterruptedLock = Lock()
-#if os(Windows)
-private var signalWatchingPipe: [HANDLE] = [INVALID_HANDLE_VALUE, INVALID_HANDLE_VALUE]
-#else
-private var signalWatchingPipe: [Int32] = [0, 0]
-private var oldAction = sigaction()
-#endif
-
 /// This class can be used by command line tools to install a handler which
 /// should be called when a interrupt signal is delivered to the process.
+@available(*, deprecated, message: "use DispatchSource instead")
 public final class InterruptHandler {
+    /// Interrupt signal handling global variables
+    private static var wasInterrupted = false
+    private static var wasInterruptedLock = Lock()
+    #if os(Windows)
+    private static var signalWatchingPipe: [HANDLE] = [INVALID_HANDLE_VALUE, INVALID_HANDLE_VALUE]
+    #else
+    private static var signalWatchingPipe: [Int32] = [0, 0]
```

Comment:
Probably better to use a tuple for this one and the windows one but meh that's really not important.